Bigtable/src/Instance.php

```diff
@@ -0,0 +1,302 @@
+use Google\Cloud\Bigtable\Admin\V2\StorageType;
+use Google\Cloud\Bigtable\Connection\ConnectionInterface;
+use Google\Cloud\Core\Exception\NotFoundException;
+use Google\Cloud\Core\Iterator\ItemIterator;
+use Google\Cloud\Core\LongRunning\LongRunningConnectionInterface;
+use Google\Cloud\Core\LongRunning\LongRunningOperation;
+use Google\Cloud\Core\LongRunning\LROTrait;
+
+/**
+ * Represents a Cloud Bigtable instance
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Bigtable\BigtableClient;
+ *
+ * $bigtable = new BigtableClient();
+ *
+ * $instance = $bigtable->instance('my-instance');
+ * ```
+ */
+class Instance
+{
+    use LROTrait;
+
+    const STORAGE_TYPE_UNSPECIFIED = StorageType::STORAGE_TYPE_UNSPECIFIED;
+    const STORAGE_TYPE_SSD = StorageType::SSD;
+    const STORAGE_TYPE_HDD = StorageType::HDD;
+
+    const INSTANCE_TYPE_UNSPECIFIED = Instance_Type::TYPE_UNSPECIFIED;
+    const INSTANCE_TYPE_PRODUCTION = Instance_Type::PRODUCTION;
+    const INSTANCE_TYPE_DEVELOPMENT = Instance_Type::DEVELOPMENT;
+
+    const STATE_TYPE_STATE_NOT_KNOWN = Instance_State::STATE_NOT_KNOWN;
+    const STATE_TYPE_READY = Instance_State::READY;
+    const STATE_TYPE_CREATING = Instance_State::CREATING;
+
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var array
+     */
+    private $info;
+
+    /**
+     * Create an object representing a Cloud Bigtable instance.
+     *
+     * @param ConnectionInterface $connection The connection to the
+     *        Cloud Bigtable Admin API.
+     * @param LongRunningConnectionInterface $lroConnection An implementation
+     *        mapping to methods which handle LRO resolution in the service.
+     * @param array $lroCallables
+     * @param string $projectId The project ID.
+     * @param string $instanceId The instance ID.
+     * @param array $info [optional] A representation of the instance object.
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        LongRunningConnectionInterface $lroConnection,
+        array $lroCallables,
+        $projectId,
+        $instanceId,
+        array $info = []
+    ) {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        
+        if (strpos($instanceId, '/') !== false) {
+            $namePrefix = 'projects/' .$projectId. '/instances/';
+            if (substr($instanceId, 0, strlen($namePrefix)) === $namePrefix) {
+                $name = $instanceId;
+            } else {
+                throw new \Exception('Instance id ' .$instanceId. ' is not formatted correctly.
+                Please use the format `my-instance` or ' .$namePrefix. 'my-instance.');
+            }
+        } else {
+            $name = $this->fullyQualifiedInstanceName($instanceId, $projectId);
```

Comment:
Is the function `fullyQualifiedInstanceName` necessary at all? (Not sure if this is suggested by someone).
It looks like just a wrapper of `instanceName()` without more business logic there.

To me, it rather adds more level of code reading unnecessarily.
Besides, you use a related util method `InstanceAdminClient::parseName()` directly below, so why don't we just `instanceName()` directly?

Reply:
Agreed... The `fullyQualifiedInstanceName` is, if I recall, an artifact carried over from Spanner from a time before the parse methods in the generated clients were fully usable.